### ISSUE TITLE
.Net Framework 3.5 (offline) installation failure is not detected

### ISSUE THREAD

**Issue by alarkvell** [brianlala/AutoSPInstaller#25]
I am using Windows Server 2016 and AutoSPInstaller 3.99.60 and have configured it for offline installation of .Net Framework 3.5. If the installation fails, AutoSPInstaller does not detect the failure and continues. For example:
```
--------------------------------------------------------------
--------------------------------------------------------------
 - Installing Prerequisite Software:
  - .Net Framework 3.5.1 from "C:\Install\2016\SharePoint\PrerequisiteInstallerFiles\sxs"...
Deployment Image Servicing and Management tool
Version: 10.0.14393.0

Image Version: 10.0.14393.0

Enabling feature(s)
[==========================100.0%==========================]

Error: 0x800f081f

The source files could not be found.
Use the "Source" option to specify the location of the files that are required to restore the feature. For more information on specifying a source location, see http://go.microsoft.com/fwlink/?LinkId=243077.

The DISM log file can be found at C:\Windows\Logs\DISM\dism.log
Done.
  - Running Prerequisite Installer (offline mode)......................
```

Would it be correct for AutoSPInstaller to stop on failure?

**Comment by CraigHumphrey** [comment]
I just got a similar error with Windows Server 2019, SP2019 and AutoSPInstaller 3.99.60.

` - Installing Prerequisite Software:
  - .Net Framework 3.5.1 from "\\ctadev03\devteam\SP19\2019\SharePoint\PrerequisiteInstallerFiles\sxs"...
Deployment Image Servicing and Management tool
Version: 10.0.17763.1

Image Version: 10.0.17763.316

Enabling feature(s)
[==========================100.0%==========================]

Error: 0x800f081f

The source files could not be found.
Use the "Source" option to specify the location of the files that are required to restore the feature. For more information on specifying a source location, see http://go.microsoft.com/fwlink/?LinkId=243077.

The DISM log file can be found at C:\Windows\Logs\DISM\dism.log
Done.`

I had `dotnetfx35sp1setup.exe` in the `2019\SharePoint\PrerequisiteInstallerFiles\sxs\` folder, but this doesn't appear to be the version that Windows Server 2019 wants.  If I run it manually, it says to use PowerShell or the Server Manager.

When I run from ServerManager, it want's a path to the install file(s) and looking at the Windows Server 2019 install media, it's looking for something quite different:
`microsoft-windows-netfx3-ondemand-package~31bf3856ad364e35~amd64~~.cab
Microsoft-Windows-NetFx3-OnDemand-Package~31bf3856ad364e35~amd64~en-US~.cab`

**Comment by brianlala** [comment]
Yep if you want to be covered for both Win2016 and Win2019 make sure these are in your \sxs\ folder:

`
microsoft-windows-netfx3-ondemand-package.cab
microsoft-Windows-NetFx3-OnDemand-Package~31bf3856ad364e35~amd64~en-US~.cab
microsoft-windows-netfx3-ondemand-package~31bf3856ad364e35~amd64~~.cab
`

Brian

**Comment by centuryx476** [comment]
> 
> 
> Yep if you want to be covered for both Win2016 and Win2019 make sure these are in your \sxs\ folder:
> 
> `microsoft-windows-netfx3-ondemand-package.cab microsoft-Windows-NetFx3-OnDemand-Package~31bf3856ad364e35~amd64~en-US~.cab microsoft-windows-netfx3-ondemand-package~31bf3856ad364e35~amd64~~.cab`
> 
> Brian

This is exactly what I was looking for to make it a Full offline Install.
Thank You
